### PR TITLE
ebasar/TSK-162-create-initial-commit-with-minimal-tasks-app

### PR DESCRIPTION
**Pull request by eemrebasar**

### CODE REVIEW

File: tasks/urls.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from django.contrib import admin
+from django.urls import path
+from django.conf.urls import url, include
+from tasks.api.v3 import urls as v3_urls
+
+
+urlpatterns = [
+    path('admin/', admin.site.urls),
+    url(r"api/v3/", include([url(r"^", include(v3_urls.api_patterns)),])),
```
Comment (LenardMarcin) [inline tasks/urls.py:9]: please keep consistency for quotes `'` `"`

**Review by LenardMarcin — COMMENTED** [review]
I think the healthcheck doesn't have to be in `/api/v3` let's just have it under bare `/healthcheck`

it's a detail anyway. good job :+1:

File: tasks/api/v3/views.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+from rest_framework import generics
+from rest_framework import status
+from rest_framework.response import Response
+
+class HealthCheckView(generics.GenericAPIView):
+
+    def get(self, request, *args, **kwargs):
+        return Response(None, status=status.HTTP_200_OK)
```
Comment (bugraaydin) [inline tasks/api/v3/views.py:8]: new line at the end of file

File: tasks/settings.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+# Application definition
+
+INSTALLED_APPS = [
+    'django.contrib.admin',
+    'django.contrib.auth',
+    'django.contrib.contenttypes',
+    'django.contrib.sessions',
+    'django.contrib.messages',
+    'django.contrib.staticfiles',
```
Comment (bugraaydin) [inline tasks/settings.py:39]: install "django_extensions" for shell plus etc

File: tasks/settings.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    'django.contrib.sessions.middleware.SessionMiddleware',
+    'django.middleware.common.CommonMiddleware',
+    'django.middleware.csrf.CsrfViewMiddleware',
+    'django.contrib.auth.middleware.AuthenticationMiddleware',
+    'django.contrib.messages.middleware.MessageMiddleware',
+    'django.middleware.clickjacking.XFrameOptionsMiddleware',
+]
+
+ROOT_URLCONF = 'tasks.urls'
+
```
Comment (bugraaydin) [inline tasks/settings.py:54]: add APPEND_SLASH = False as it is our convention

File: tasks/api/v3/urls.py [hunk 5]
```diff
@@ -0,0 +1,10 @@
+from django.urls import include
+from tasks.api.v3.views import HealthCheckView
+from django.conf.urls import url
+
+
+api_patterns = [
+    url(r'^healthcheck/',
```
Comment (bugraaydin) [inline tasks/api/v3/urls.py:7]: no trailing slash

File: tests/tasks/api_v3/view_tests.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+class HealthCheckViewTests(APITestCase):
+
+    def setUp(self):
+        self.client = Client()
+        self.url = reverse("api-v3-health-check")
+
+    def test_healthcheck_response_ok(self):
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, 200)
```
Comment (bugraaydin) [inline tests/tasks/api_v3/view_tests.py:17]: new line at eof

File: tests/tasks/api_v3/view_tests.py [hunk 7]
```diff
@@ -0,0 +1,17 @@
+
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.test import APITestCase
+from django.test import Client
+from django.conf import settings
```
Comment (bugraaydin) [inline tests/tasks/api_v3/view_tests.py:6]: unused import

File: tests/tasks/api_v3/view_tests.py [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+
+
+class HealthCheckViewTests(APITestCase):
+
+    def setUp(self):
+        self.client = Client()
```
Comment (bugraaydin) [inline tests/tasks/api_v3/view_tests.py:12]: no need for this in a apitestcase

**Review by LenardMarcin — APPROVED** [review]
looks good, thank you :+1:

File: tasks/settings.py [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+                'django.template.context_processors.debug',
+                'django.template.context_processors.request',
+                'django.contrib.auth.context_processors.auth',
+                'django.contrib.messages.context_processors.messages',
+            ],
+        },
+    },
+]
+
+WSGI_APPLICATION = 'tasks.wsgi.application'
```
Comment (rdidyk) [inline tasks/settings.py:73]: so, wsgi or asgi ?

File: tasks/urls.py [hunk 10]
```diff
@@ -0,0 +1,10 @@
+from django.contrib import admin
+from django.urls import path
+from django.conf.urls import url, include
+from tasks.api.v3 import urls as v3_urls
+
+
+urlpatterns = [
+    path('admin/', admin.site.urls),
+    url(r'api/v3/', include([url(r'^', include(v3_urls.api_patterns)),])),
```
Comment (rdidyk) [inline tasks/urls.py:9]: is it normal that we start from the third version right away?

**Review by rdidyk — APPROVED** [review]

File: tasks/urls.py (hunk 10 above)
Comment (bugraaydin) [reply tasks/urls.py:9]: @rdidyk we thought that v3 convention would be across all plentific services, not service specific, are we right on that? we might be wrong :D
i agree with @LenardMarcin that healthcheck view shouldnt belong to v3 but rather just `/health-check` tho

**Pull request merged**